packages/typespec-ts/test/modularIntegration/generated/azure/example/basic/src/index.d.ts

```diff
@@ -0,0 +1,43 @@
+import { ClientOptions } from '@azure-rest/core-client';
+import { OperationOptions } from '@azure-rest/core-client';
+import { Pipeline } from '@azure/core-rest-pipeline';
+
+export declare interface ActionRequest {
+    stringProperty: string;
+    modelProperty?: Model;
+    arrayProperty?: string[];
+    recordProperty?: Record<string, string>;
+}
+
+export declare interface ActionResponse {
+    stringProperty: string;
+    modelProperty?: Model;
+    arrayProperty?: string[];
+    recordProperty?: Record<string, string>;
+}
+
+export declare class AzureExampleClient {
+    private _client;
+    readonly pipeline: Pipeline;
+    constructor(options?: AzureExampleClientOptionalParams);
+    basicAction(queryParam: string, headerParam: string, body: ActionRequest, options?: BasicActionOptionalParams): Promise<ActionResponse>;
+}
+
+export declare interface AzureExampleClientOptionalParams extends ClientOptions {
+    apiVersion?: string;
+}
+
+export declare interface BasicActionOptionalParams extends OperationOptions {
+}
+
+export declare type Enum = "EnumValue1";
+
+export declare interface Model {
+    int32Property?: number;
+    float32Property?: number;
+    enumProperty?: Enum;
+}
+
+export declare type Versions = "2022-12-01-preview";
+
+export { }
```

Comment:
check why this happened?

Reply:
i think this is a convetion to let compiler know this is a module not script and you could see: https://www.typescriptlang.org/docs/handbook/2/modules.html#how-javascript-modules-are-defined.